src/mavsdk/core/system_impl.cpp

```diff
@@ -532,7 +562,7 @@ void SystemImpl::set_connected()
     }
     if (enable_needed) {
         if (has_autopilot()) {
-            send_autopilot_version_request();
+            // send_autopilot_version_request();
```

Comment:
Hmm that's weird :thinking:. Though I think it's needed, right?

Reply:
I am not sure. But it could be that before adding the result callback to the `send_autopilot_version_request() ` it was basically acting like an async call. So it was exiting quickly. But now since we wait for the result its blocking for some time? 

On the other side i cannot figure it out if sending the autopilot version request is needed for anything which comes after. Maybe you can comment on that?

Reply:
The following is the output with the autopilot version command DISABLED.
```
[09:39:51|Debug] Callback debugging is on. (mavsdk_impl.cpp:25)
Connection Success. 
Waiting to discover system...
[09:39:51|Info ] New system on: 127.0.0.1:54634 (with system ID: 1) (udp_connection.cpp:192)
[09:39:51|Debug] New system ID: 1 Comp ID: 1 (mavsdk_impl.cpp:601)
[09:39:51|Debug] Component Autopilot (1) added. (system_impl.cpp:376)
[09:39:51|Debug] Discovered 1 component(s) (system_impl.cpp:530)
Discovered autopilot
Got System. 
[09:39:51|Warn ] command unsupported (528). (mavlink_command_sender.cpp:202)
[09:39:51|Warn ] Trying alternative command (512).. (system_impl.cpp:517)
```
The following is the output with the autopilot version command ENABLED.

```
[09:55:14|Debug] Callback debugging is on. (mavsdk_impl.cpp:25)
Connection Success. 
Waiting to discover system...
[09:55:14|Info ] New system on: 127.0.0.1:54634 (with system ID: 1) (udp_connection.cpp:192)
[09:55:14|Debug] New system ID: 1 Comp ID: 1 (mavsdk_impl.cpp:601)
[09:55:14|Debug] Component Autopilot (1) added. (system_impl.cpp:376)
[09:55:14|Debug] Discovered 1 component(s) (system_impl.cpp:530)
[09:55:14|Warn ] sending again after 0.501881 s, retries to do: 3  (520). (mavlink_command_sender.cpp:291)
[09:55:15|Warn ] Callback called from mavsdk_impl.cpp:617 took more than 1 second to run. (mavsdk_impl.cpp:707)
Aborted (core dumped)
```

Any clues here?

Reply:
created `send_autopilot_version_request_async()` version so that it will not block in the `void SystemImpl::set_connected()` function. I am still not sure if we care about the success of this message in this function or it is more left to the plugins to take care of that. At least in the Info plugin the message is sent again on init.